### PR TITLE
LSC job for provisioning default JMAP identity

### PR DESCRIPTION
**Pull request by quantranhong1999**
need more tests, and documentation, switch to the latest tmail image when possible.

### CODE REVIEW

File: src/test/java/org/lsc/plugins/connectors/james/JamesIdentityDstServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+    private static final URL PUBLIC_KEY = ClassLoader.getSystemResource("conf/jwt_publickey");
+    private static final int JAMES_WEBADMIN_PORT = 8000;
+    private static final boolean FROM_SAME_SERVICE = true;
+
+    private GenericContainer<?> james;
+    private JamesIdentityDstService testee;
+
+    @BeforeEach
+    void setup() throws Exception {
+        james = new GenericContainer<>("quanth99/tmail-backend-memory:lsc-identity-2");
```
Comment (Arsnael) [inline src/test/java/org/lsc/plugins/connectors/james/JamesIdentityDstServiceTest.java:69]: Let's not forget to put it back to master when the necessary stuff has been merged :)

**Review by Arsnael — COMMENTED** [review]
Read it

File: src/test/java/org/lsc/plugins/connectors/james/integration/IdentitySynchronizationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    private RSAPrivateKey getPrivateKey() throws Exception {
+        try (PEMReader pemReader = new PEMReader(new FileReader(PRIVATE_KEY.getFile()), "james"::toCharArray)) {
+            Object readObject = pemReader.readObject();
+            return (RSAPrivateKey) ((java.security.KeyPair) readObject).getPrivate();
+        }
+    }
+
+    // TODO write test running the lsc job connect TMail and LDAP
+    @Disabled("This is for manual test now. Could give it more effort to be a automation test")
+    @Test
```
Comment (quantranhong1999) [inline src/test/java/org/lsc/plugins/connectors/james/integration/IdentitySynchronizationIntegrationTest.java:138]: Manual test (local):
Given we have Alice and Bob entries in LDAP.
Then LSC identity job should create default identity for Alice and BOB:
- LSC:
image
- James get identity for BOB:
image

**Comment by quantranhong1999** [comment]
The code is ready to be reviewed.
Manual testing locally seems good.
I would put some more effort to set up an integration test for LSC.

**Review by Arsnael — COMMENTED** [review]
Read it. You have tests with ldap in james I believe that you could look at right? Good job overall :)

File: src/test/java/org/lsc/plugins/connectors/james/JamesIdentityDstServiceTest.java (hunk 1 above)
Comment (chibenwa) [reply src/test/java/org/lsc/plugins/connectors/james/JamesIdentityDstServiceTest.java:70]: Pro tip: => Create a dedicated issue in the sprint backlog

**Review by chibenwa — COMMENTED** [review]
Looks good so far. I think we have the time-budget to invest on this automated test...

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**